apps/guardian-ui/src/components/CustomStepper.tsx

```diff
@@ -0,0 +1,167 @@
+import React from 'react';
+import { Box, Center, Flex, Text } from '@chakra-ui/react';
+import { ReactComponent as CheckIcon } from '../assets/svgs/white-check.svg';
+import { StepState } from '../types';
+
+interface StepProps {
+  text: string;
+  state: StepState;
+  margin?: string;
+}
+
+const Step: React.FC<StepProps> = ({ text, state, margin = '-48px' }) => {
+  const active = state === StepState.Active;
+  const inActive = state === StepState.InActive;
+  const completed = state === StepState.Completed;
+
+  const colorState = (
+    inActiveColor: string,
+    activeColor: string,
+    completedColor: string
+  ) => (inActive ? inActiveColor : active ? activeColor : completedColor);
+
+  const rectangleState = () =>
+    state === StepState.InActive ? '#EAECF0' : '#1570ef';
+
+  return (
+    <Flex
+      w='100%'
+      flexDir='column'
+      alignItems='flex-end'
+      h='72px'
+      justifyContent='space-between'
+    >
+      <Flex alignItems='center' h='100%' w='100%'>
+        <Box w='100%' h='2px' bgColor={rectangleState()}></Box>
+        <Flex>
+          <Center
+            borderRadius={'50%'}
+            h={active ? '36px' : '24px'}
+            w={active ? '36px' : '24px'}
+            bgColor={colorState('transparent', '#EFF8FF', 'transparent')}
+          >
+            <Center
+              borderRadius='50%'
+              h='24px'
+              w='24px'
+              bgColor={colorState('#F2F4F7', '#1570ef', '#1570ef')}
+            >
+              {completed ? (
+                <CheckIcon />
+              ) : (
+                <Center
+                  borderRadius='50%'
+                  h='8px'
+                  w='8px'
+                  bgColor='white'
+                ></Center>
+              )}
+            </Center>
+          </Center>
+        </Flex>
+      </Flex>
+      <Text
+        textTransform='capitalize'
+        fontWeight='600'
+        fontSize='14px'
+        color={colorState('#344054', '#1570ef', '#1570ef')}
+        whiteSpace='nowrap'
+        mr={margin}
+      >
+        {text}
+      </Text>
+    </Flex>
+  );
+};
+
+const InitialStep: React.FC<Pick<StepProps, 'state'>> = ({ state }) => {
```

Comment:
why is the initial step special? Can we avoid having a special initial step?

Reply:
Based on the Figma design and the design structure used to achieve the stepper, we need an initial step state. Making it to be optional from the design structure of the step component, will make the initial step to be disjointed from the rest of the design